diff -durN freedt-0.18.orig/iolib.c freedt-0.18/iolib.c
--- freedt-0.18.orig/iolib.c	2003-05-27 20:06:16.000000000 +0200
+++ freedt-0.18/iolib.c	2004-02-04 14:26:18.408747224 +0100
@@ -212,50 +212,50 @@
 }
 
 /* Format an argument into the buffer. */
-static void format_arg(buffer *b, char t, va_list *va) {
+static void format_arg(buffer *b, char t, va_list va) {
 	switch (t) {
 	case 'c': {
-		char *s = va_arg(*va, char *);
+		char *s = va_arg(va, char *);
 		bappends(b, s);
 		break;
 	}
 	case 'b': {
-		buffer *src = va_arg(*va, buffer *);
+		buffer *src = va_arg(va, buffer *);
 		bappend(b, src);
 		break;
 	}
 	case 'i': {
-		int i = va_arg(*va, int);
+		int i = va_arg(va, int);
 		format_long(b, i, 10);
 		break;
 	}
 	case 'I': {
-		unsigned int i = va_arg(*va, unsigned int);
+		unsigned int i = va_arg(va, unsigned int);
 		format_ulong(b, i, 10);
 		break;
 	}
 	case 'x': {
-		int i = va_arg(*va, int);
+		int i = va_arg(va, int);
 		format_long(b, i, 16);
 		break;
 	}
 	case 'X': {
-		unsigned int i = va_arg(*va, unsigned int);
+		unsigned int i = va_arg(va, unsigned int);
 		format_ulong(b, i, 16);
 		break;
 	}
 	case 'l': {
-		long i = va_arg(*va, long);
+		long i = va_arg(va, long);
 		format_long(b, i, 10);
 		break;
 	}
 	case 'L': {
-		unsigned long i = va_arg(*va, unsigned long);
+		unsigned long i = va_arg(va, unsigned long);
 		format_ulong(b, i, 10);
 		break;
 	}
 	case 'a': {
-		unsigned long i = va_arg(*va, unsigned long);
+		unsigned long i = va_arg(va, unsigned long);
 		format_long(b, i & 0xff, 10);
 		bappendc(b, '.');
 		format_long(b, (i >> 8) & 0xff, 10);
@@ -285,7 +285,7 @@
 		++p;
 		if (*p == '\0')
 			die("unexpected end of format");
-		format_arg(b, *p, &va);
+		format_arg(b, *p, va);
 
 		fmt = p + 1;
 	}
